.github/workflows/tests.yml

```diff
@@ -96,31 +96,39 @@ jobs:
 
       - name: Pull and cache Docker images
         if: steps.cache-docker.outputs.cache-hit != 'true'
+        shell: bash
         run: |
-          # Extract image versions from docker config
-          WAVS_IMAGE=$(grep 'WAVS_DOCKER_IMAGE:' taskfile/docker.yml | cut -d"'" -f2 | cut -d'{' -f1)
-          FOUNDRY_IMAGE=$(grep 'FOUNDRY_DOCKER_IMAGE:' taskfile/docker.yml | cut -d'"' -f2)
-          MIDDLEWARE_IMAGE=$(grep 'MIDDLEWARE_DOCKER_IMAGE:' taskfile/docker.yml | cut -d"'" -f2 | cut -d'{' -f1)
-          JAEGER_IMAGE=$(grep 'JAEGER_DOCKER_IMAGE:' taskfile/docker.yml | cut -d'"' -f2)
-          PROMETHEUS_IMAGE=$(grep 'PROMETHEUS_DOCKER_IMAGE:' taskfile/docker.yml | cut -d'"' -f2)
-
-          # Create cache directory
+          set -euo pipefail
           mkdir -p /tmp/docker-cache
 
-          # Pull and save images
+          # Map YAML vars -> output filenames
+          declare -A FILES=(
+            ["WAVS_DOCKER_IMAGE"]="wavs"
+            ["FOUNDRY_DOCKER_IMAGE"]="foundry"
+            ["MIDDLEWARE_DOCKER_IMAGE"]="middleware"
+            ["JAEGER_DOCKER_IMAGE"]="jaeger"
+            ["PROMETHEUS_DOCKER_IMAGE"]="prometheus"
+            ["POA_MIDDLEWARE_DOCKER_IMAGE"]="poa_middleware"
+          )
+
+          YAML_FILE="taskfile/docker.yml"
+
           echo "Pulling and caching Docker images..."
-          docker pull $WAVS_IMAGE
-          docker pull $FOUNDRY_IMAGE  
-          docker pull $MIDDLEWARE_IMAGE
-          docker pull $JAEGER_IMAGE
-          docker pull $PROMETHEUS_IMAGE
-
-          # Save images to cache
-          docker save $WAVS_IMAGE -o /tmp/docker-cache/wavs.tar
-          docker save $FOUNDRY_IMAGE -o /tmp/docker-cache/foundry.tar
-          docker save $MIDDLEWARE_IMAGE -o /tmp/docker-cache/middleware.tar
-          docker save $JAEGER_IMAGE -o /tmp/docker-cache/jaeger.tar
-          docker save $PROMETHEUS_IMAGE -o /tmp/docker-cache/prometheus.tar
+          for VAR_NAME in "${!FILES[@]}"; do
```

Comment:
this section looks a little complex... isn't there a Taskfile-native way to do it cleaner? Just a curiosity/comment though, not a blocker as long as things work